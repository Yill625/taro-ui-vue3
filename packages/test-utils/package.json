{
  "name": "@taro-ui-vue3/test-utils",
  "version": "0.0.0",
  "main": "dist/index.js",
  "license": "MIT",
  "dependencies": {
    "vue": "^3.2.0",
    "@vue/test-utils": "^2.0.0-rc.0"
  }
}