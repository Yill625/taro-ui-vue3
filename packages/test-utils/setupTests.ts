import toHaveBeenWarnedInit from './been-warned'

toHaveBeenWarnedInit()