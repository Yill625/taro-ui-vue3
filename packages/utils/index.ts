import propsFactory from './propsFactory'

export * from './common'
export {
  propsFactory
}