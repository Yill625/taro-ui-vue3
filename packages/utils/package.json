{
  "name": "@taro-ui-vue3/utils",
  "version": "0.0.0",
  "main": "dist/index.js",
  "license": "MIT",
  "peerDependencies": {
    "@tarojs/taro": "^3.3.0",
    "vue": "^3.2.0"
  },
  "devDependencies": {
    "@vue/test-utils": "^2.0.0-rc.0"
  }
}