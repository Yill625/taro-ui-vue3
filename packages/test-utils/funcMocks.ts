import * as utils from "@taro-ui-vue3/utils/common"
import { pxTransformMockFn, getEnvMockFn } from './@tarojs/taro'

let uuid: jest.SpyInstance
let getEnvs: jest.SpyInstance
let pxTransform: jest.SpyInstance

function initMocks() {
  beforeEach(() => {
    jest.mock('@taro-ui-vue3/utils/common', () => {
      return utils
    })

    pxTransform = jest
      .spyOn(utils, 'pxTransform')
      .mockImplementation(pxTransformMockFn)

    uuid = jest
      .spyOn(utils, 'uuid')
      .mockReturnValue('__tests__2021')

    getEnvs = jest
      .spyOn(utils, 'getEnvs')
      .mockImplementation(() => {
        const env = getEnvMockFn()
        return {
          isWEAPP: env === 'WEAPP',
          isALIPAY: env === 'ALIPAY',
          isWEB: env === 'WEB'
        }
      })
  })

  afterEach(() => {
    uuid.mockRestore()
    getEnvs.mockRestore()
    pxTransform.mockRestore()
  })
}

export default initMocks