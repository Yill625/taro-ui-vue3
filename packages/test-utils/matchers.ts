import * as utils from "@taro-ui-vue3/utils/common"

jest.mock('@taro-ui-vue3/utils/common')

let warn: jest.SpyInstance
let uuid: jest.SpyInstance

const asserted: Set<string> = new Set()
const NOOP = () => { }

// Ensure console.warn is called with expected message
const toHaveBeenWarned: jest.CustomMatcher = (received: string) => {
  asserted.add(received)
  const passed = warn.mock.calls.some(args => args[0].indexOf(received) > -1)
  if (passed) {
    return {
      pass: true,
      message: () => `expected "${received}" not to have been warned.`
    }
  } else {
    const msgs = warn.mock.calls.map(args => args[0]).join('\n - ')
    return {
      pass: false,
      message: () =>
        `expected "${received}" to have been warned` +
        (msgs.length
          ? `.\n\nActual messages:\n\n - ${msgs}`
          : ` but no warning was recorded.`)
    }
  }
}

// Ensure previous console.warn is called with expected message
const toHaveBeenWarnedLast: jest.CustomMatcher = (received: string) => {
  asserted.add(received)
  const passed =
    warn.mock.calls[warn.mock.calls.length - 1][0].indexOf(received) > -1
  if (passed) {
    return {
      pass: true,
      message: () => `expected "${received}" not to have been warned last.`
    }
  } else {
    const msgs = warn.mock.calls.map(args => args[0]).join('\n - ')
    return {
      pass: false,
      message: () =>
        `expected "${received}" to have been warned last.\n\nActual messages:\n\n - ${msgs}`
    }
  }
}

// Ensure console.warn is called n times with expected message
const toHaveBeenWarnedTimes: jest.CustomMatcher = (received: string, n: number) => {
  asserted.add(received)
  let found = 0
  warn.mock.calls.forEach(args => {
    if (args[0].indexOf(received) > -1) {
      found++
    }
  })

  if (found === n) {
    return {
      pass: true,
      message: () => `expected "${received}" to have been warned ${n} times.`
    }
  } else {
    return {
      pass: false,
      message: () =>
        `expected "${received}" to have been warned ${n} times but got ${found}.`
    }
  }
}

function initWarnMatchers() {
  beforeEach(() => {
    asserted.clear()
    warn = jest.spyOn(console, 'warn').mockImplementation(NOOP)
    uuid = jest.spyOn(utils, 'uuid').mockReturnValue('__tests__2021')

    // NOT working in concurrent mode
    // should avoid use in test.concurrent()
    expect.extend({
      toHaveBeenWarned,
      toHaveBeenWarnedLast,
      toHaveBeenWarnedTimes
    })
  })

  afterEach(() => {
    const assertedArray = Array.from(asserted)
    const nonAssertedWarnings = warn.mock.calls
      .map(args => args[0])
      .filter(received => {
        return !assertedArray.some(assertedMsg => {
          return received.indexOf(assertedMsg) > -1
        })
      })

    uuid.mockRestore()
    warn.mockRestore()

    if (nonAssertedWarnings.length) {
      throw new Error(
        `test case threw unexpected warnings:\n - ${nonAssertedWarnings.join(
          '\n - '
        )}`
      )
    }
  })
}

export default initWarnMatchers