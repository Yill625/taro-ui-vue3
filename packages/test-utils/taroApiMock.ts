import * as utils from "@taro-ui-vue3/utils/common"
import Taro from '@tarojs/taro'
import { createAnimation } from '@tarojs/taro-h5'

import {
  getEnvMockFn,
  pxTransformMockFn,
  chooseImageMockFn,
  querySelectorMockFn
} from './@tarojs/taro'

Taro.pxTransform = pxTransformMockFn
Taro.chooseImage = chooseImageMockFn
Taro.createSelectorQuery = querySelectorMockFn()
Taro.createAnimation = createAnimation

let uuid: jest.SpyInstance
let getEnvs: jest.SpyInstance
let getenv: jest.SpyInstance
let pxTransform: jest.SpyInstance
let chooseImage: jest.SpyInstance

function initTaroMockApis() {
  beforeEach(() => {
    jest.mock('@tarojs/taro', () => {
      return Taro
    })

    jest.mock('@taro-ui-vue3/utils/common', () => {
      return utils
    })

    getenv = jest
      .spyOn(Taro, 'getEnv')
      .mockImplementation(getEnvMockFn)

    chooseImage = jest
      .spyOn(Taro, 'chooseImage')
      .mockImplementation(chooseImageMockFn)

    pxTransform = jest
      .spyOn(utils, 'pxTransform')
      .mockImplementation(pxTransformMockFn)

    uuid = jest
      .spyOn(utils, 'uuid')
      .mockReturnValue('__tests__2021')

    getEnvs = jest
      .spyOn(utils, 'getEnvs')
      .mockImplementation(() => {
        const env = getEnvMockFn()
        return {
          isWEAPP: env === 'WEAPP',
          isALIPAY: env === 'ALIPAY',
          isWEB: env === 'WEB'
        }
      })
  })

  afterEach(() => {
    uuid.mockRestore()
    getEnvs.mockRestore()
    getenv.mockRestore()
    pxTransform.mockRestore()
    chooseImage.mockRestore()
  })
}

export default initTaroMockApis
