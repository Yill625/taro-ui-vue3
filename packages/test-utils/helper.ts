import { mount, shallowMount, DOMWrapper, VueWrapper } from '@vue/test-utils'
import { VNode, Slot } from '@vue/runtime-core'

export const sleep = async (timeout: number): Promise<null> =>
  new Promise((resolve) => setTimeout(resolve, timeout))

export interface Slots {
  default: Array<string | VNode> | Slot
  [x: string]: Array<string | VNode> | Slot
}

export function mountFactory(
  comp: any,
  components?: Record<string, object>,
  props?: Record<string, any>,
  slots?: Slots
) {
  return mount<any>(comp, {
    global: {
      components
    },
    props,
    slots,
  })
}

export function genMountFn(
  comp: any,
  components?: Record<string, object>,
  useShallowMount?: boolean
) {
  return (props = {}, slots?: Slots) => {
    return useShallowMount
      ? shallowMountFactory(comp, components, props, slots)
      : mountFactory(comp, components, props, slots)
  }
}

export function shallowMountFactory(
  comp: any,
  components?: Record<string, object>,
  props?: Record<string, any>,
  slots?: Slots
) {
  return shallowMount<any>(comp, {
    global: {
      components
    },
    props,
    slots,
  })
}

export interface Dimension {
  top?: number
  bottom?: number
  left?: number
  right?: number
  width?: number
  height?: number
}

export function genDelayedSelectorSpy(
  utils: any,
  dimensions: Dimension
): jest.SpyInstance {
  return jest
    .spyOn(utils, 'delayQuerySelector')
    .mockImplementation((_, sel: string, delay?: number) => {
      return new Promise((resolve) => {
        resolve([dimensions])
      })
    })
}

export async function triggerTouchEvents(
  el: DOMWrapper<Element>,
  touchstartDetails: any,
  touchendDetails: any
) {
  const startDetails = {
    touches: [touchstartDetails],
    changedTouches: [touchstartDetails]
  }

  const endDetails = {
    touches: [touchendDetails],
    changedTouches: [touchendDetails]
  }

  await el.trigger('touchstart', startDetails)
  await el.trigger('touchmove', endDetails)
  await el.trigger('touchend', endDetails)
}

export function testPropClassAndStyle(
  mountFn: (props?: {}, slots?: Slots | undefined) => VueWrapper<any>
) {

  it.concurrent('should render prop class', async () => {
    const wrapper = mountFn({ class: 'test' })
    expect(
      wrapper.classes()
    ).toContain('test')
  })

  it.concurrent('should render prop style', async () => {
    const wrapper = mountFn({ style: { color: 'red' } })
    expect(
      wrapper.attributes('style')
    ).toContain('color: red;')
  })
}

export function hexToRGBA(hex: string, opacity = 1) {
  hex = hex.replace('#', '')
  const rgb = hex
    .match(new RegExp('(.{' + hex.length / 3 + '})', 'g'))!
    .map((l) => parseInt(hex.length % 2 ? l + l : l, 16))

  const rgba = rgb.concat(isFinite(opacity) ? opacity : 1)

  return opacity === 1
    ? `rgb(${rgb.join(', ')})`
    : `rgba(${rgba.join(', ')})`
}