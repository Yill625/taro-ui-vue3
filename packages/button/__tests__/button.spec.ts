import { mountFactory, Slots } from '@taro-ui-vue3/test-utils/helper'
import AtButton from '../index'
import * as utils from '@taro-ui-vue3/utils/common'

const factory = (
  values = {},
  slots: Slots = { default: ['按钮'] }
) => {
  return mountFactory(AtButton, {}, values, slots)
}

describe('AtButton', () => {
  it('should render prop -- size(normal)', () => {
    const wrapper = factory({
      size: 'normal',
    })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('should render prop -- size(small)', () => {
    const wrapper = factory({
      size: 'small',
    })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('should render prop -- type(primary)', () => {
    const wrapper = factory({
      type: 'primary',
    })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('should render prop -- type(secondary)', () => {
    const wrapper = factory({
      type: 'secondary',
    })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('should render prop -- circle', () => {
    const wrapper = factory({
      circle: true,
    })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('should render prop -- full', () => {
    const wrapper = factory({
      full: true,
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- loading', async () => {
    const wrapper = factory({
      loading: true,
    })
    expect(wrapper.element).toMatchSnapshot()

    await wrapper.setProps({ loading: false })
    expect(wrapper.find('.at-button__icon').exists()).toBe(false)
  })

  it('should render prop -- formType', async () => {
    const wrapper = factory({
      formType: 'submit',
    })
    expect(wrapper.find('.at-button__wxbutton').attributes('formtype')).toBe('submit')

    await wrapper.setProps({ formType: '' })
    wrapper.vm.$nextTick(() => {
      expect(wrapper.find('.at-button__wxbutton').attributes('formtype')).toBe('')
    })
  })

  describe('should render prop -- disabled', () => {
    beforeEach(() => {
      jest.mock('@taro-ui-vue3/utils/common')
    })

    it('should not render any Button element when disabled', () => {
      const wrapper = factory({
        disabled: true,
      })
      expect(wrapper.element).toMatchSnapshot()
      expect(wrapper.find('.at-button__wxbutton').exists()).toBe(false)
    })

    it('should render weapp Button', () => {
      const getEnvs = jest.spyOn(utils, 'getEnvs').mockImplementation(() => {
        return {
          isWEAPP: true,
          isALIPAY: false,
          isWEB: false
        }
      })
      const wrapper = factory()
      expect(wrapper.element).toMatchSnapshot()
      expect(wrapper.find('form > .at-button__wxbutton').exists()).toBe(true)

      getEnvs.mockClear()
    })

    it('should render web button', () => {
      const getEnvs = jest.spyOn(utils, 'getEnvs').mockImplementation(() => {
        return {
          isWEAPP: false,
          isALIPAY: false,
          isWEB: true
        }
      })
      const wrapper = factory()
      expect(wrapper.element).toMatchSnapshot()
      expect(wrapper.find('.at-button > .at-button__wxbutton').exists()).toBe(true)
      getEnvs.mockClear()
    })

    it('should render alipay Button', () => {
      const getEnvs = jest.spyOn(utils, 'getEnvs').mockImplementation(() => {
        return {
          isWEAPP: false,
          isALIPAY: true,
          isWEB: false
        }
      })
      const wrapper = factory()
      expect(wrapper.element).toMatchSnapshot()
      expect(wrapper.find('.at-button > .at-button__wxbutton').exists()).toBe(true)
      getEnvs.mockClear()
    })
  })
})

describe('AtButton Behavior', () => {
  it('AtButton should trigger onClick event', () => {
    const onClick = jest.fn()
    const wrapper = factory({
      onClick: onClick,
    })
    wrapper.find('.at-button').trigger('tap')
    expect(onClick).toBeCalled()
  })

  it('AtButton should not trigger onClick event when disabled', () => {
    const onClick = jest.fn()
    const wrapper = factory({
      disabled: true,
      onClick: onClick,
    })
    wrapper.find('.at-button').trigger('tap')
    expect(onClick).not.toBeCalled()
  })

  it('AtButton should not trigger onClick event if handler not passed', () => {
    const onClick = jest.fn()
    const wrapper = factory()
    wrapper.find('.at-button').trigger('tap')
    expect(onClick).not.toBeCalled()
  })

  describe('AtButton should trigger Taro button events', () => {
    let getEnvs
    beforeEach(() => {
      jest.mock('@taro-ui-vue3/utils/common')
      getEnvs = jest.spyOn(utils, 'getEnvs').mockImplementation(() => {
        return {
          isWEAPP: true,
          isALIPAY: false,
          isWEB: false
        }
      })
    })

    afterEach(() => {
      getEnvs.mockClear()
    })

    it('should trigger onGetUserInfo when openType == getUserInfo', () => {
      const onGetUserInfo = jest.fn()
      const wrapper = factory({
        openType: 'getUserInfo',
        onGetUserInfo: onGetUserInfo
      })
      wrapper.find('.at-button__wxbutton').trigger('getuserinfo')
      expect(onGetUserInfo).toBeCalled()
    })

    it('should not trigger onGetUserInfo when openType != getUserInfo', () => {
      const onGetUserInfo = jest.fn()
      const wrapper = factory({
        onGetUserInfo: onGetUserInfo
      })
      wrapper.find('.at-button__wxbutton').trigger('getuserinfo')
      expect(onGetUserInfo).not.toBeCalled()
    })

    it('should not trigger onGetUserInfo if eventHandler not passed', () => {
      const onGetUserInfo = jest.fn()
      const wrapper = factory({
        openType: 'getUserInfo'
      })
      wrapper.find('.at-button__wxbutton').trigger('getuserinfo')
      expect(onGetUserInfo).not.toBeCalled()
    })

    it('should trigger onContact when openType == contact', () => {
      const onContact = jest.fn()
      const wrapper = factory({
        openType: 'contact',
        onContact: onContact
      })
      wrapper.find('.at-button__wxbutton').trigger('contact')
      expect(onContact).toBeCalled()
    })

    it('should not trigger onContact if eventHandler not passed', () => {
      const onContact = jest.fn()
      const wrapper = factory({
        openType: 'contact'
      })
      wrapper.find('.at-button__wxbutton').trigger('contact')
      expect(onContact).not.toBeCalled()
    })

    it('should not trigger onContact when openType != contact', () => {
      const onContact = jest.fn()
      const wrapper = factory({
        onContact: onContact
      })
      wrapper.find('.at-button__wxbutton').trigger('contact')
      expect(onContact).not.toBeCalled()
    })

    it('should trigger onOpenSetting when openType == openSetting', () => {
      const onOpenSetting = jest.fn()
      const wrapper = factory({
        openType: 'openSetting',
        onOpenSetting: onOpenSetting
      })
      wrapper.find('.at-button__wxbutton').trigger('opensetting')
      expect(onOpenSetting).toBeCalled()
    })

    it('should not trigger onOpenSetting if eventHandler not passed', () => {
      const onOpenSetting = jest.fn()
      const wrapper = factory({
        openType: 'openSetting'
      })
      wrapper.find('.at-button__wxbutton').trigger('opensetting')
      expect(onOpenSetting).not.toBeCalled()
    })

    it('should not trigger onOpenSetting when openType != openSetting', () => {
      const onOpenSetting = jest.fn()
      const wrapper = factory({
        onOpenSetting: onOpenSetting
      })
      wrapper.find('.at-button__wxbutton').trigger('opensetting')
      expect(onOpenSetting).not.toBeCalled()
    })

    it('should trigger onGetPhoneNumber when openType == getPhoneNumber', () => {
      const onGetPhoneNumber = jest.fn()
      const wrapper = factory({
        openType: 'getPhoneNumber',
        onGetPhoneNumber: onGetPhoneNumber
      })
      wrapper.find('.at-button__wxbutton').trigger('getphonenumber')
      expect(onGetPhoneNumber).toBeCalled()
    })

    it('should trigger onGetPhoneNumber if eventHandler not passed', () => {
      const onGetPhoneNumber = jest.fn()
      const wrapper = factory({
        openType: 'getPhoneNumber'
      })
      wrapper.find('.at-button__wxbutton').trigger('getphonenumber')
      expect(onGetPhoneNumber).not.toBeCalled()
    })

    it('should not trigger onGetPhoneNumber when openType != getPhoneNumber', () => {
      const onGetPhoneNumber = jest.fn()
      const wrapper = factory({
        onGetPhoneNumber: onGetPhoneNumber
      })
      wrapper.find('.at-button__wxbutton').trigger('getphonenumber')
      expect(onGetPhoneNumber).not.toBeCalled()
    })

    it('should trigger onLaunchapp and onError when openType == launchApp', () => {
      const onLaunchapp = jest.fn()
      const onError = jest.fn()
      const wrapper = factory({
        openType: 'launchApp',
        onLaunchapp: onLaunchapp,
        onError: onError
      })
      wrapper.find('.at-button__wxbutton').trigger('launchapp')
      wrapper.find('.at-button__wxbutton').trigger('error')
      expect(onLaunchapp).toBeCalled()
      expect(onError).toBeCalled()
    })

    it('should trigger onLaunchapp and onError if eventHandlers not passed', () => {
      const onLaunchapp = jest.fn()
      const onError = jest.fn()
      const wrapper = factory({
        openType: 'launchApp'
      })
      wrapper.find('.at-button__wxbutton').trigger('launchapp')
      wrapper.find('.at-button__wxbutton').trigger('error')
      expect(onLaunchapp).not.toBeCalled()
      expect(onError).not.toBeCalled()
    })

    it('should not trigger onLaunchapp and onError when openType != launchApp', () => {
      const onLaunchapp = jest.fn()
      const onError = jest.fn()
      const wrapper = factory({
        onLaunchapp: onLaunchapp,
        onError: onError
      })
      wrapper.find('.at-button__wxbutton').trigger('launchapp')
      wrapper.find('.at-button__wxbutton').trigger('error')
      expect(onLaunchapp).not.toBeCalled()
      expect(onError).not.toBeCalled()
    })

    it('should trigger onGetAuthorize when openType == getAuthorize', () => {
      getEnvs = jest.spyOn(utils, 'getEnvs').mockImplementation(() => {
        return {
          isWEAPP: false,
          isALIPAY: true,
          isWEB: false
        }
      })
      const onGetAuthorize = jest.fn()
      const wrapper = factory({
        openType: 'getAuthorize',
        onGetAuthorize: onGetAuthorize
      })
      wrapper.find('.at-button__wxbutton').trigger('getauthorize')
      expect(onGetAuthorize).toBeCalled()
    })

    it('should not trigger onGetAuthorize when openType != getAuthorize', () => {
      getEnvs = jest.spyOn(utils, 'getEnvs').mockImplementation(() => {
        return {
          isWEAPP: false,
          isALIPAY: true,
          isWEB: false
        }
      })
      const onGetAuthorize = jest.fn()
      const wrapper = factory({
        onGetAuthorize: onGetAuthorize
      })
      wrapper.find('.at-button__wxbutton').trigger('getauthorize')
      expect(onGetAuthorize).not.toBeCalled()
    })

    it('should not trigger onGetAuthorize if eventHandler not passed', () => {
      getEnvs = jest.spyOn(utils, 'getEnvs').mockImplementation(() => {
        return {
          isWEAPP: false,
          isALIPAY: true,
          isWEB: false
        }
      })
      const onGetAuthorize = jest.fn()
      const wrapper = factory({ openType: 'getAuthorize' })
      wrapper.find('.at-button__wxbutton').trigger('getauthorize')
      expect(onGetAuthorize).not.toBeCalled()
    })

  })
})
