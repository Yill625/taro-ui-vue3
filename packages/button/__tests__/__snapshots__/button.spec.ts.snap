// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtButton should render prop -- circle 1`] = `
<view
  class="at-button--normal at-button--circle at-button"
>
  <button
    class="at-button__wxbutton"
    lang="en"
  />
  <!---->
  <!---->
  <!---->
  <view
    class="at-button__text"
  >
    按钮
  </view>
</view>
`;

exports[`AtButton should render prop -- disabled should not render any Button element when disabled 1`] = `
<view
  class="at-button--normal at-button--disabled at-button"
>
  <!---->
  <!---->
  <!---->
  <!---->
  <view
    class="at-button__text"
  >
    按钮
  </view>
</view>
`;

exports[`AtButton should render prop -- disabled should render alipay Button 1`] = `
<view
  class="at-button--normal at-button"
>
  <!---->
  <!---->
  <button
    class="at-button__wxbutton"
    formtype=""
    lang="en"
    showmessagecard="false"
  />
  <!---->
  <view
    class="at-button__text"
  >
    按钮
  </view>
</view>
`;

exports[`AtButton should render prop -- disabled should render weapp Button 1`] = `
<view
  class="at-button--normal at-button"
>
  <!---->
  <form>
    <button
      class="at-button__wxbutton"
      formtype=""
      lang="en"
      showmessagecard="false"
    />
  </form>
  <!---->
  <!---->
  <view
    class="at-button__text"
  >
    按钮
  </view>
</view>
`;

exports[`AtButton should render prop -- disabled should render web button 1`] = `
<view
  class="at-button--normal at-button"
>
  <button
    class="at-button__wxbutton"
    lang="en"
  />
  <!---->
  <!---->
  <!---->
  <view
    class="at-button__text"
  >
    按钮
  </view>
</view>
`;

exports[`AtButton should render prop -- full 1`] = `
<view
  class="at-button--normal at-button--full at-button"
>
  <button
    class="at-button__wxbutton"
    lang="en"
  />
  <!---->
  <!---->
  <!---->
  <view
    class="at-button__text"
  >
    按钮
  </view>
</view>
`;

exports[`AtButton should render prop -- loading 1`] = `
<view
  class="at-button--normal at-button--icon at-button"
>
  <button
    class="at-button__wxbutton"
    lang="en"
  />
  <!---->
  <!---->
  <view
    class="at-button__icon"
  >
    <view
      class="at-loading"
    >
      <view
        class="at-loading__ring"
      />
      <view
        class="at-loading__ring"
      />
      <view
        class="at-loading__ring"
      />
    </view>
  </view>
  <view
    class="at-button__text"
  >
    按钮
  </view>
</view>
`;

exports[`AtButton should render prop -- size(normal) 1`] = `
<view
  class="at-button--normal at-button"
>
  <button
    class="at-button__wxbutton"
    lang="en"
  />
  <!---->
  <!---->
  <!---->
  <view
    class="at-button__text"
  >
    按钮
  </view>
</view>
`;

exports[`AtButton should render prop -- size(small) 1`] = `
<view
  class="at-button--small at-button"
>
  <button
    class="at-button__wxbutton"
    lang="en"
  />
  <!---->
  <!---->
  <!---->
  <view
    class="at-button__text"
  >
    按钮
  </view>
</view>
`;

exports[`AtButton should render prop -- type(primary) 1`] = `
<view
  class="at-button--normal at-button--primary at-button"
>
  <button
    class="at-button__wxbutton"
    lang="en"
  />
  <!---->
  <!---->
  <!---->
  <view
    class="at-button__text"
  >
    按钮
  </view>
</view>
`;

exports[`AtButton should render prop -- type(secondary) 1`] = `
<view
  class="at-button--normal at-button--secondary at-button"
>
  <button
    class="at-button__wxbutton"
    lang="en"
  />
  <!---->
  <!---->
  <!---->
  <view
    class="at-button__text"
  >
    按钮
  </view>
</view>
`;
